presslabs/wordpress-integration
# Setup separate pool for cron

In order to make sure that each time a wp-cron.php request hits the php-fpm backend, the cron will work, we need to separate the workers in different pools.
For that, we'll have two workers, on that will do the actual request and the other one that will answer if another request is still processing, via the Wordpress mechanisms.

We'll also need to set the user and group to www-data. 

Related to https://github.com/presslabs/wordpress-operator/issues/9

#### Tests - green
- build the docker image with `WORDPRESS_VERSION` set to `dev-master` since it contains the fixed Wordpress patches. 
```
docker build --build-arg WORDPRESS_VERSION=dev-master -t presslabs/wordpress-integration .
```
- `docker run -p 8080:80 presslabs/wordpress-integration`
- `curl -SI localhost:8080/wp/wp-cron.php`
- you **should** receive 
```
HTTP/1.1 200 OK
Server: openresty
Date: Wed, 06 Mar 2019 17:56:42 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
X-Powered-By: PHP/7.2.15
```

#### Tests - red
- edit `docker/templates/php-fpm.d/www.conf` and set `cron` pool listen to `9003` (patch bellow).
```patch
diff --git a/docker/templates/php-fpm.d/www.conf b/docker/templates/php-fpm.d/www.conf
index bafdc16..1461728 100644
--- a/docker/templates/php-fpm.d/www.conf
+++ b/docker/templates/php-fpm.d/www.conf
@@ -74,7 +74,7 @@ pm = static
 pm.max_children = {{ div $max_children 2 | add1 }}

 [cron]
-listen = 9002
+listen = 9003

 user = www-data
 group = www-data
``` 
- build the docker image with `WORDPRESS_VERSION` set to `dev-master` since it contains the fixed Wordpress patches. 
```
docker build --build-arg WORDPRESS_VERSION=dev-master -t presslabs/wordpress-integration .
```
- `docker run -p 8080:80 presslabs/wordpress-integration`
- you **should** receive 
```
HTTP/1.1 502 Bad Gateway
Server: openresty
Date: Thu, 07 Mar 2019 07:17:54 GMT
Content-Type: text/html
Content-Length: 170
Connection: keep-alive
```
